workitem/link/link_repository_blackbox_test.go

```diff
@@ -61,6 +61,33 @@ func (s *linkRepoBlackBoxTest) TestList() {
 	})
 }
 
+func (s *linkRepoBlackBoxTest) TestChildrenOrderOfExecution() {
+	// tests order of workitem children returned by list is according to the
+	// "order of execution" specified
+	s.T().Run("ok - child work items order", func(t *testing.T) {
+		fxt := tf.NewTestFixture(t, s.DB,
+			tf.WorkItems(4), // parent + child 1-3
+			tf.WorkItemLinkTypes(1, func(fxt *tf.TestFixture, idx int) error {
+				fxt.WorkItemLinkTypes[idx].ForwardName = "parent of"
+				return nil
+			}),
+			tf.WorkItemLinks(3, func(fxt *tf.TestFixture, idx int) error {
+				fxt.WorkItemLinks[idx].SourceID = fxt.WorkItems[0].ID
+				fxt.WorkItemLinks[idx].TargetID = fxt.WorkItems[idx+1].ID
+				return nil
+			}),
+		)
+
+		res, _, err := s.workitemLinkRepo.ListWorkItemChildren(s.Ctx, fxt.WorkItems[0].ID, nil, nil)
+		require.NoError(t, err)
+		require.Len(t, res, 3)
+
+		assert.Equal(t, fxt.WorkItems[1].Fields[workitem.SystemOrder], res[2].Fields[workitem.SystemOrder].(float64))
+		assert.Equal(t, fxt.WorkItems[2].Fields[workitem.SystemOrder], res[1].Fields[workitem.SystemOrder].(float64))
+		assert.Equal(t, fxt.WorkItems[3].Fields[workitem.SystemOrder], res[0].Fields[workitem.SystemOrder].(float64))
```

Comment:
Can we make it easy to read by doing -
```
expectedOder := []float64{ value1, value2, value3,}
for i, v := range expectedOrder {
   assert.Equal(t, v, res[i].Fields[workitem.SystemOrder].(float64))
}
```
IMO this is more verbose than earlier
OR
**golden files** :)

Reply:
@pranavgore09 I will add the above change.
Also, as we disucussed, I will add the tests for reordering the workitem children.

Reply:
@pranavgore09 I have optimized the code in this commit -> 30df9ca
Also, I have added test that says "list children in OOE after updating their OOE" in these two commits 9a83979 and 30fcfb0